### ISSUE TITLE
NcSelect ARIA fixes 

### ISSUE THREAD

**Issue by emoral435** [nextcloud-libraries/nextcloud-vue#4900]
## Expected Behavior

The NcSelect component should behave like [this input](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-both/) does for ARIA accessibility. Whenever the user types something, show the top closest matches, and when the user exits the input selector, save the top closest match to the input list field

## Actual Behavior
As of now, the input component just saves whatever the user selects down. We can implement functionality to have the top result be chosen, perhaps as either the default option or it being a setting that happens whenever a combobox boolean prop is passed through, for specification.

## To Do

- Decide on whether or not this should be passed as a boolean combobox prop or just part of the default functionality
- Implement this similarly to the links above specifications

## Related issue links

- [first](https://github.com/nextcloud/server/issues/41902)
- [second](https://github.com/nextcloud/server/issues/41909)
- [third](https://github.com/nextcloud/server/issues/41908)
- and so on, there are a lot of issues on the main servers repo that are due to this

**Comment by emoral435** [comment]
for some reason, I cannot assign myself to this issue, but consider myself assigned to this issue as I assigned myself this issue on the main server repo

**Comment by emoral435** [comment]
nextcloud/server#41902 is what this issue relates too, and because we are working against that issue, this issue is now no longer needed.
